File: src/main/java/com/cae/mapped_exceptions/specifics/InputMappedException.java
package com.cae.mapped_exceptions.specifics;


import com.cae.mapped_exceptions.MappedException;

public class InputMappedException extends MappedException {

    public InputMappedException(String briefPublicMessage, String details){
        super(briefPublicMessage, details);
    }

    public InputMappedException(String briefPublicMessage){
        super(briefPublicMessage);
    }

    public InputMappedException(String briefPublicMessage, String details, Exception originalException){
        super(briefPublicMessage, details, originalException);
    }

    public InputMappedException(String briefPublicMessage, Exception originalException){
        super(briefPublicMessage, originalException);
    }


}

File: implementations/python/src/cae_mapped_exceptions/specifics/input_mapped_exception.py

from cae_mapped_exceptions.mapped_exception import MappedException

class InputMappedException(MappedException):

    def __init__(self, brief_public_message: str | None = None, details: str | None = None, original_exception: Exception  | None = None):
        super().__init__(brief_public_message, details, original_exception)